source/server/listener_impl.cc

```diff
@@ -377,6 +378,9 @@ void ListenerImpl::buildUdpListenerFactory(Network::Socket::Type socket_type,
     ProtobufTypes::MessagePtr message =
         Config::Utility::translateToFactoryConfig(udp_config, validation_visitor_, config_factory);
     udp_listener_factory_ = config_factory.createActiveUdpListenerFactory(*message, concurrency);
+
+    udp_listener_worker_router_ =
```

Comment:
Can we make it nullptr if prefer_quic_kernel_bpf_packet_routing is on? Thus skipping registering and unregistering if udpListenerWorkerRouter() returns nullopt

Reply:
We could, but it would make it more complicated for very little gain. Listeners are created/removed infrequently, and the memory cost is low compared to the overall footprint of a listener, so I don't feel that it's worth it.